drivers: i2c: gecko: Fix EFM32 I2C target

edt-alex opened the pull request:
Fix compile error due to misnamed macro.

Fix response to callback behavior so NACKs are issued if the callback returns an error. This allows for proper signaling to the I2C host when commands or data are invalid for the I2C target.

jerome-pouiller commented on drivers/i2c/i2c_gecko.c:
```diff
@@ -192,7 +192,7 @@ static int i2c_gecko_target_register(const struct device *dev, struct i2c_target
 	I2C_SlaveAddressSet(config->base, cfg->address << _I2C_SADDR_ADDR_SHIFT);
 	/* Match with specified address, no wildcards in address */
-	I2C_SlaveAddressMaskSet(config->base, _I2C_SADDRMASK_SADDRMASK_MASK);
+	I2C_SlaveAddressMaskSet(config->base, _I2C_SADDRMASK_MASK);
```
I believe this change is only cosmetics. Could you place it in a separated commit (thus, the reviewers can do a quick on check cosmetics commit and focus on a smaller commit for the important changes).

edt-alex replied:
No this change is not just for cosmetics, this is the "Fix compile error due to misnamed macro" change I mentioned. 

I failed to mention that I'm building against a SiLabs Thunderboard EFM32GG12B ([SLTB009A](https://www.silabs.com/development-tools/mcu/32-bit/thunderboard-gg12-kit)).

I've looked deeper just now and it seems like maybe the EF**R**32 part headers [define the old](https://github.com/zephyrproject-rtos/hal_silabs/blob/190a144a16bed9a938a94543ed5bbc70c0552e0f/simplicity_sdk/platform/Device/SiliconLabs/EFR32FG23/Include/efr32fg23_i2c.h#L458) macro but the EF**M**32 part I'm using [does not](https://github.com/zephyrproject-rtos/hal_silabs/blob/190a144a16bed9a938a94543ed5bbc70c0552e0f/gecko/Device/SiliconLabs/EFM32GG12B/Include/efm32gg12b_i2c.h#L330). The `_I2C_SADDRMASK_MASK` I'm suggesting to change to is defined in both part headers.

edt-alex replied:
Done. 

I'm a new contributor, so let me know if my commit splits are as you would prefer.

edt-alex replied:
oops, pushing again to reflow my commit messages

jerome-pouiller commented on drivers/i2c/i2c_gecko.c:
```diff
@@ -303,23 +325,23 @@ void i2c_gecko_isr(const struct device *dev)
+	I2C_DEVICE_DT_INST_DEFINE(idx, i2c_gecko_init, NULL, &i2c_gecko_data_##idx,                \
+				  &i2c_gecko_config_##idx, POST_KERNEL, CONFIG_I2C_INIT_PRIORITY,  \
+				  &i2c_gecko_driver_api);                                          \
+                                                                                                   \
```
I also believe this change is only cosmetics.

edt-alex replied:
Done

jerome-pouiller commented on drivers/i2c/i2c_gecko.c:
```diff
@@ -233,6 +233,7 @@ void i2c_gecko_isr(const struct device *dev)
 {
 	const struct i2c_gecko_config *config = dev->config;
 	struct i2c_gecko_data *data = dev->data;
+	int cb_ret = 0;
```
`ret` is probably descriptive enough. I believe this change will help to help to keep code in 100 columns (the real fix would be to split this function, but this is out of the scope of this PR).

I also suggest to only initialize variable during declaration if it is useful (it is easier for the reader and some static analysis tools can detect misuse of the variable).

edt-alex replied:
Done

jerome-pouiller commented on drivers/i2c/i2c_gecko.c:
```diff
@@ -249,31 +250,52 @@ void i2c_gecko_isr(const struct device *dev)
-				config->base->TXDATA = tx_byte;
+				cb_ret = data->target_cfg->callbacks->read_requested(
+					data->target_cfg, &tx_byte);
+				if (0 == cb_ret) {
```
We try to avoid "Yoda notation". Prefer `if (ret == 0)` (or even `if (ret)`)

edt-alex replied:
Done

jerome-pouiller reviewed (commented):
Only nitpicks. otherwise, looks good to me.

edt-alex committed:
drivers: i2c: gecko: Fix EFM32 I2C target

Fix compile error due to unknown macro when building CONFIG_I2C_TARGET for
Silabs Thunderboard EFM32GG12 (SLTB009A). The previous macro exists in
(all?) EFR32 part headers but not EFM32.

Signed-off-by: Alex Hogen <alex@edt.com>

edt-alex committed:
drivers: i2c: gecko: Fix EFM32 I2C target

Fix I2C target implementation so NACKs are issued if the callback returns
an error. This allows for proper signaling to the I2C host when commands
or data are invalid for the I2C target.

Signed-off-by: Alex Hogen <alex@edt.com>

edt-alex committed:
drivers: i2c: gecko: Fix EFM32 I2C target

clang-format on drivers/i2c/i2c_gecko.c

Signed-off-by: Alex Hogen <alex@edt.com>
